tools/src/compileProtos.ts

```diff
@@ -344,14 +345,16 @@ export async function generateRootName(directories: string[]): Promise<string> {
   // It's OK to play some guessing game here: if we locate `package.json`
   // with a package name, we'll use it; otherwise, we'll fallback to 'default'.
   for (const directory of directories) {
-    const packageJson = path.resolve(directory, '..', 'package.json');
-    if (fs.existsSync(packageJson)) {
-      const json = JSON.parse((await readFile(packageJson)).toString()) as {
-        name: string;
-      };
-      const name = json.name.replace(/[^\w\d]/g, '_');
-      const hopefullyUniqueName = `${name}_protos`;
-      return hopefullyUniqueName;
+    for (const p of walkUp(path.resolve(directory, '..'))) {
```

Comment:
I'm concerned in cases where the package is deeply nested (we use this in google-cloud-node, which has at least two layers of package.json). Could you add a test that makes sure we find the closest package.json, and stop once we've found it?

Reply:
Actually the existing test already tests it. There's package.json in tools, and another one in the project root.

I adapted the test, so it passes now, and added another test in `{os.tmpdir()}/protoList/empty` (none of them actually exists, but it shouldn't matter), which will really fall-back to default.